modules/openy_gc_auth/modules/openy_gc_auth_daxko_sso/src/Controller/DaxkoLinkController.php

```diff
@@ -133,22 +134,30 @@ public function backlink(Request $request) {
     // Based on user token, get logged user data.
     $userData = $this->daxkoClient->getMyInfo($token);
 
-    $userDetails = $this->daxkoClient->getRequest('members/' . $userData->member_id);
-    // Check if this user is an active client.
-    if ($userDetails->active) {
-      // Create drupal user if it doesn't exist and login it.
-      $name = $userDetails->name->first_name . ' ' . $userDetails->name->last_name;
-      $email = $userDetails->emails[0]->email;
-
-      // Authorize user (register, login, log, etc).
-      $this->gcUserAuthorizer->authorizeUser($name, $email);
-
-      return new RedirectResponse($this->configFactory->get('openy_gated_content.settings')->get('virtual_y_url'));
-    }
-    else {
-      // Redirect back to Virual Y login page.
-      return new RedirectResponse($this->configFactory->get('openy_gated_content.settings')->get('virtual_y_login_url'));
+    if ($userData) {
+      $userUnitData = $this->daxkoClient->getRequest('/units/' . $userData->member_unit_id);
+      $userDetails = $this->daxkoClient->getRequest('members/' . $userData->member_id);
+      // Check if this user is an active client.
+      if ($userUnitData->status == 'Active') {
+        // Create drupal user if it doesn't exist and login it.
+        $name = $userDetails->name->first_name . ' ' . $userDetails->name->last_name;
+        $email = "daxko-{$userData->member_id}@virtualy.openy.org";
```

Comment:
This change from above:
`      $email = $userDetails->emails[0]->email;`
results in a php Error:
`Drupal\Core\Entity\EntityStorageException: SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry 'Test User-en' for key 'user__name': `
I was able to resolve it by deleting all of the existing user accounts and having them re-created fresh, but that might not be a great option for everyone.

Reply:
Good catch.
I think, we have to write an hook_update and delete existed virtual-y users from the system.

Reply:
@froboy, i've added post update that deletes all virtual_y user + added member id to the user name.
There should not be such problem after that.